### ISSUE TITLE
[Fonts API] Do not print fonts with 'admin_print_styles' for block themes

### ISSUE THREAD

**Issue by hellofromtonya** [WordPress/gutenberg#50064]
Part of the Fonts API Roadmap #41479.

### Description

tl;dr

When using the block editor / site editor, fonts are being printed twice:

* One time by `'admin_print_styles'` hook to insert into the web page's main doc. <- this should not happen.
* Second time in the iframed editor. <- this should happen.

Longer:

In the 6.2 compat script loader (`lib/compat/wordpress-6.2/script-loader.php`), the Fonts API was added to `gutenberg_resolve_assets_override()` to wire printing enqueued fonts for insertion into the iframed editor. Though not backported to Core as part of 6.2, this functionality was added for testing and preparation for Core introduction.

But the printing wired to `'admin_print_styles'` action remained. The problem is: the `@font-face` styles are being generated twice.

The hook to `'admin_print_styles'` should be for classic powered sites, not block powered sites.

```php
add_action( 'admin_print_styles', 'wp_print_fonts', 50 );
```

Initialization logic is needed to determine if printing should be hooked into  'admin_print_styles'.

### Step-by-step reproduction instructions

1. Use WordPress 6.2+
2. Activate TT3 theme.
3. Activate the Gutenberg plugin.
4. Open the Site Editor.
5. In your browser, open Dev Tools to the `Inspector` tab.
6. Search for `"wp-fonts-local"`.

Notice there are 2 instances of this element: one in the main doc's `<head>` and the other in the iframe.

### Screenshots, screen recording, code snippet

![FontsAPI-print2x](https://user-images.githubusercontent.com/7284611/234310421-8f4c1432-6020-4c17-a810-177b94e609e4.gif)

### Environment info

- WordPress 6.2
- Theme: TT3
- Gutenberg: 15.6.2 and `trunk`

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
